cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -100,19 +109,68 @@ pub enum Error {
     Network(#[from] network::Error),
     #[error(transparent)]
     Wasm(#[from] wasm::Error),
+    #[error("alias \"{alias}\" already exist. Use --force to override it.")]
+    AliasAlreadyExist { alias: String },
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
+        self.validate_alias()?;
+
         let res = self.run_against_rpc_server(None, None).await?.to_envelope();
         match res {
             TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
             TxnEnvelopeResult::Res(contract) => {
+                self.save_contract_id(&contract);
                 println!("{contract}");
             }
         }
         Ok(())
     }
+
+    fn validate_alias(&self) -> Result<(), Error> {
+        let alias = self.alias();
+
+        if alias.is_empty() {
+            return Ok(());
+        }
+
+        let path = self.alias_path();
+
+        if path.exists() && !self.force {
+            Err(Error::AliasAlreadyExist { alias })
+        } else {
+            Ok(())
+        }
+    }
+
+    fn alias(&self) -> String {
+        match &self.alias {
+            Some(n) => n.to_string(),
+            None => String::new(),
+        }
+    }
+
+    fn alias_path(&self) -> PathBuf {
+        let config_dir = self.config.config_dir().unwrap();
+        let name = format!("{}.txt", self.alias());
+
+        config_dir.join("contract-ids").join(name)
+    }
```

Comment:
```suggestion
    fn alias_path(&self) -> Result<Option<PathBuf>, Error> {
        let config_dir = self.config.config_dir()?;
        Ok(self.alias.as_ref().map(|alias| config_dir.join("contract-ids").join(name)))
    }
```
We already added the errors from config to this files `Error` enum. This means we can use `?` if we return a `Result`. Then we can also return an `Option` in case there was no alias. Also I suggest removing the `.txt`.